probe/sniff/sniffer_test.go

```diff
@@ -77,7 +72,16 @@ func TestMerge(t *testing.T) {
 			},
 		},
 		NodeMetadatas: report.NodeMetadatas{
-			srcEndpointNodeID: report.MakeNodeMetadata(),
+			srcEndpointNodeID: {
+				Metadata:  map[string]string{},
+				Counters:  map[string]int{},
+				Adjacency: report.MakeIDList(dstEndpointNodeID),
+			},
+			dstEndpointNodeID: report.NodeMetadata{
+				Metadata:  map[string]string{},
+				Counters:  map[string]int{},
+				Adjacency: report.MakeIDList(),
+			},
```

Comment:
We could do the method cascading AKA fluent API:
```
MakeNodeMetadata().SetMetadata(...).SetCounters(...).SetAdjacency(...)
```
I'm not a huge fan of this, but it beats the N^2 constructors we'd need otherwise.